### PR TITLE
improve testing

### PR DESCRIPTION
**Pull request by micgro42**
Improve the test coverage, so that we can refactor and add new features more confidently.

### CODE REVIEW

File: _test/field_yesno.test.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+{
+
+    public function dataProvider()
+    {
+        return [
+            [
+                'cb:@@checkbox@@',
+                '"=Yes"',
+                '"!No"',
+                '1',
```
Comment (dregad) [inline _test/field_yesno.test.php]: Shouldn't that be `Yes` ?

>default checked substitution
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'cb:Yes'
+'cb:1'

File: _test/field_yesno.test.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                'cb:Yes',
+                [],
+                'default checked substitution',
+            ],
+            [
+                'cb:@@checkbox@@',
+                '"=Yes"',
+                '"!No"',
+                '0',
```
Comment (dregad) [inline _test/field_yesno.test.php]: Shouldn't that be `No` ?

>default unchecked substitution
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'cb:No'
+'cb:0'

Comment (Klap-in) [reply _test/field_yesno.test.php]: Indeed! There is already a pull request submitted to fix this, see #234

Comment (dregad) [reply _test/field_yesno.test.php]: @Klap-in thanks... Guess I should have looked at 7b3bab9462a22bcd83e10a389edd1a8d1fc1e432 commit message :flushed:

**Pull request merged**